### ISSUE TITLE
Test bdk performance with bigger wallets

### ISSUE THREAD

**Issue by danielabrozzoni** [bitcoindevkit/bdk#1346]

**Comment by oleonardolima** [comment]
I've been wondering and thinking about this one, and some questions arose:

1. Are we looking to benchmark the performance against which other wallets, e.g. Bitcoin Core, Sparrow, Nunchuk?

2. Is there any previous art on this topic? All I found was related to syncing, (maybe I didn't research enough). 

3. Also, what would be an ideal (real-world) scenario for creating workloads? For example, is the one mentioned in #985 a decent one?

**Comment by notmandatory** [comment]
I'm not aware of any prior art for wallet test performance. But we could start with picking our own small, medium, large, xlarge wallet sizes, say 10, 100, 1000, 10_000 tx respectively and see how long Wallet::transactions take for each.  Once we have this test we can add variations like adding in RBFs and such later.

**Comment by oleonardolima** [comment]
Agree, it looks like a good initial approach. I'll try to ask around too to check if anybody knows about some useful libraries/tools, or existing workloads.

Also, it looks like a common pain point, after discussing it with StratumV2 contributor during SatsConf.

I've only used something similar in Clojure, for data-specified testing/workload production in a web-based environment (e.g. https://github.com/metosin/malli), will need to check an equivalent in Rust.

**Comment by nymius** [comment]
> Observation from @matthiasdebernardini is slowness with `Wallet::transactions` which calls function `TxGraph::list_canonical_txs`. We should profile this with a large number of tx and different mixes of confirmed and unconfirmed transactions.

To keep the references together, d4102b4b6bb885934a3bfd8efffeb550541efe7f makes advances in this direction.
